src/rf/js/src/home/components/modals.js

```diff
@@ -107,9 +188,224 @@ var UploadModal = React.createBackboneClass({
         this.dragCounter = 0;
     },
 
+    getLayerFields: function() {
+        var self = this,
+            getTextValue = function(refName) {
+                return React.findDOMNode(self.refs[refName]).value;
+            },
+            getSelectValue = function(refName) {
+                var node = React.findDOMNode(self.refs[refName]);
+                return node.options[node.selectedIndex].value;
+            },
+            getChecked = function(refName) {
+                return React.findDOMNode(self.refs[refName]).checked;
+            },
+            defaultColor = '#FFFFFF',
+            layerFields = {
+                name: getTextValue('name'),
+                organization: getTextValue('organization'),
+                description: getTextValue('description'),
+                tags: $(React.findDOMNode(self.refs['tags'])).tagsinput('items'),
+                capture_start: getTextValue('capture_start'),
+                capture_end: getTextValue('capture_end'),
+                area: getTextValue('area'),
+                area_units: getTextValue('area_units'),
+                projection: '4325', //getSelectValue('projection'),
+                srid: '4325', //getTextValue('srid'),
+                tile_srid: getSelectValue('tile_srid'),
+                tile_format: getSelectValue('tile_format'),
+                resampling: getSelectValue('resampling'),
+                transparency: getChecked('transparency') ?
+                    defaultColor : getTextValue('transparency_color'),
+                tile_origin: getChecked('tile_origin') ?
+                    'topleft' : 'bottomleft'
+            };
+
+        return layerFields;
+    },
+
+    getClientErrors: function() {
+        var errors = this.getInitialClientErrors();
+
+        if (this.state.fileDescriptions.length === 0) {
+            errors.files = 'Please select at least one file.';
+        }
+
+        if (!this.state.name) {
+            errors.name = 'Please enter a name.';
+        } else if (this.state.name.length > 255) {
+            errors.name = 'Please enter a name < 255 characters.';
+        }
+
+        if (!this.state.organization) {
+            errors.organization = 'Please enter an organization.';
+        } else if (this.state.organization.length > 255) {
+            errors.organization = 'Please enter an organization < 255 characters.';
+        }
+
+        if (!this.state.description) {
+            errors.description = 'Please enter a description.';
+        }
+
+        function setDateErrors(state, errors, fieldName) {
+            if (!state[fieldName]) {
+                errors[fieldName] = 'Please enter a valid date.';
+            } else {
+                var date = moment(state[fieldName]);
+                if (date.year() < 1800) {
+                    errors[fieldName] = 'Please enter a year after 1500';
```

Comment:
Consider removing arbitrary restrictions. Also, the error message says `1500` but the code says `1800`.